testsuite/classlibrary/TestNode.sc

```diff
@@ -50,4 +53,53 @@ TestNode : UnitTest {
 		node.free;
 	}
 
+	test_releaseMsg {
+		var msg;
+
+		node = Synth(\default);
+		server.sync;
+
+		msg = node.releaseMsg(nil);
+		this.assertEquals(msg[3], 0, "nil should set gate to 0");
+
+		msg = node.releaseMsg(0);
+		this.assertEquals(msg[3], -1, "0 should set gate to -1");
+
+		msg = node.releaseMsg(2);
+		this.assertEquals(msg[3], -3, "2 should set gate to -3");
+
+		msg = node.releaseMsg(-2);
+		this.assertEquals(msg[3], -1, "A negative value should set gate to -1");
+
+		node.free;
+	}
```

Comment:
these should all be separate tests

Reply:
I don't understand. I though we wanted to avoid booting the server needlessly in order to save time running the test suite. Separating these would mean one boot/quit per test. 

Unfortunately, the server must be booted in order to test `releaseMsg`.

Reply:
I think this is how it should be writen if trying to avoid multiple asserts:
```supercollider
node = Synth(\default);
server.sync;

gates = [nil, 0, 2, -2].collect { |releaseTime|
    var msg = node.releaseMsg(releaseTime);
    msg[3]
};

this.assertEquals(gates, [0, -1, -3, -1], "...some message");
node.free;
```

Reply:
> Unfortunately, the server must be booted in order to test releaseMsg.

I don't think so; you can do this without booting the server, by mocking a Node with `Node.basicNew`. In fact, that is the only constructor `Node` offers, so IMO we should prefer using that here.

In fact this would make an excellent case for how to do mock-object tests.

Reply:
no need to boot. E.g. if you call `g = Group(s)` you get `Group(1000)` without a booted server.

Reply:
Ok - I'll rewrite.

Reply:
The server need not be booted to test whether `releaseMsg` is building the right OSC message. If you want to test whether the server is evaluating the negative gate correctly, then it would need to be booted. But perhaps that's an EnvGen test, not a releaseMsg test. What is the proper scope of this test?